src/ui/scanpopup.cc

```diff
@@ -169,8 +169,7 @@ ScanPopup::ScanPopup( QWidget * parent,
                return;
              }
              if ( cfg.preferences.pronounceOnLoadPopup ) {
-
-               definition->openLink( QUrl::fromEncoded( audioUrl.toUtf8() ), {} );
+               definition->playAudio( QUrl::fromEncoded( audioUrl.toUtf8() ) );
```

Comment:
I think delay the execution of playing first audio on JavaScript's side is better solution https://github.com/xiaoyifang/goldendict-ng/issues/1396#issuecomment-2334096514

Because the `audioUrl` here could be invalid

1. Load page
2. runJavascript → will try to find audio when the page is not fully loaded (the gdAudio map is empty)
3. Yields no `audioUrl`

Reply:
the auto-pronounce does not use runJavascript

Reply:
Ok, I probably misunderstand something 😅